araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+
+        # Determine the "reference channel" to base the CSW around as the
+        #   channel with the maximum voltage
+        max_voltage = {'voltage': -1, 'channel':-123456}
+        for ch_ID in channels_to_csw:
+            this_max_voltage = np.max(wavepacket['waveforms'][ch_ID].GetY())
+            if this_max_voltage > max_voltage['voltage']:
+                max_voltage['voltage'] = this_max_voltage
+                max_voltage['channel'] = ch_ID 
+        reference_ch = max_voltage['channel']
+
+        # Get arrival delays relative to the reference channel based on
+        #   expected arrival times from reconstruction results
+        arrival_delays_reco = self.__get_arrival_delays_reco(
+            data, reco_results, channels_to_csw, reference_ch, 
+            which_distance, solution)
+        
+        # Get arrival delays by zooming in on the cross correlation between
+        #   each channel and the reference channel around the expected
+        #   arrival delay from reconstruction results
+        if useful_event.isSoftwareTrigger(): wf_type = 'soft'
+        elif useful_event.isCalpulserEvent(): wf_type = 'cp'
+        else: wf_type = 'rf'
+        arrival_delays = self.__get_arrival_delays_AraRoot_xcorr(
+            wavepacket, polarization, channels_to_csw, reference_ch, 
+            arrival_delays_reco, wf_type, which_distance)  
+
+        # Initialize the final CSW waveform time and voltage arrays
+        csw_values = np.zeros((1, csw_length))
+        csw_times = self.__trim_array(
+            np.asarray(wavepacket['waveforms'][reference_ch].GetX()), 
+            np.asarray(wavepacket['waveforms'][reference_ch].GetY()), 
+            wavepacket['waveforms'][reference_ch].GetN()-csw_length
+        )[0]
+        csw_dt = csw_times[1] - csw_times[0]
+
+        # Roll the waveform from each channel so the starting time of each
+        #   waveform lines up. Then add the waveform to the CSW.
+        for ch_ID in channels_to_csw: 
+
+            # Load this channel's voltage and time arrays. Shift time by arrival delay
+            values = np.asarray(wavepacket['waveforms'][ch_ID].GetY())
+            times = np.asarray(wavepacket['waveforms'][ch_ID].GetX()) - arrival_delays[ch_ID]
+
+            # Determine if the time binning of this channel matches the time binning
+            #   of the CSW and adjust if it doesn't. 
+            # Ex: If the CSW has times of [0.9, 1.4, 1.9] then a channel
+            #   with a time array of [1.6, 2.1, 2.6] needs to be rebinned/shifted 
+            #   backwards by 0.2 but a channel with the time array [3.4, 3.9, 4.4 ] 
+            #   doesn't because each bin starts and ends on the same fraction of a 
+            #   nanosecond that the CSW does. 
+            rebinning_shift = round( (times[0] - csw_times[0]) % csw_dt, 4) 
+            if rebinning_shift != 0: 
+                waveform_model = Akima1DInterpolator(times, values)
```

Comment:
Maybe `waveform_interp` is clearer? But you can ignore.

Reply:
Renamed